Adjusting high-AR for Speed and Aim

Magnus-Cosmos opened the pull request:
This is a partial solution to the AR issue I previously addressed, issue #46

Note, need to see how this would affect leaderboards (it probably won't change that much), but I actually suck at coding so if someone could please recalculate the top100 or top1000, that'd be a huge help.

Currently, high-AR (and low-AR) only affects the aim component of pp. This is essentially saying that high-AR does not affect streaming (or AR in general). Put this into a thought experiment, where you take popular stream maps and convert them to AR11. From my experience, and from the trends of AR11 plays, streaming with AR11 is harder (subjective) and is underrated/not preferred (objective). The best example of a high AR streaming playstyle is [Matrix](https://osu.ppy.sh/u/5052899). Some of his +(HD),HR,DT scores on stream heavy maps are extremely impressive (subjective) and you barely see any other players replicating his plays, while on popular AR11 maps (e.g. Bombs Away, Shiori, etc.), there have been plenty of FCs.

Possible changes to this (just some ideas), aim could be nerfed less, while speed ar bonus is decreased, this way it serves mainly as a buff to speed, rather than a rebalance (though I feel like doing this might cause extra buffing to every high AR play, and won't make high AR streaming as appealing).

Basically, this adjustment/rebalance moves part of the high AR bonus from aim to speed. Aim is still affected by high AR, but (subjective) it's affected less than speed.

And here's a spreadsheet with filsdelama with the speed bonus and high-AR applied: https://docs.google.com/spreadsheets/d/1nxRwsL0CJ4SX_zr0Ec-eN6YjIH51EKn6JUaVPyNKZvA/edit#gid=40620073

Simply said, this high-AR rebalance will not affect most people. (HD)DT scores on the higher end are slightly affected because they go above AR10.33, but changes are minimal/unnoticeable. This mainly affects (HD),HR,DT scores like filsdelama's plays and rarer plays like some of Matrix's HR,DT scores.

New rankings with adjustment: http://sakisan.be/osu/pr59.html
As I predicted, most players experience little change, from a quick skim, it looks like a few (like 10-20) players lose over 100 ranks, and barely anyone gains any significant ranks (except Matrix), which basically shows why I said streaming AR11 was underrated/not preferred.

(from old dump, 05/17/18)
Hastebin of filsdelama's top40 scores: https://hastebin.com/ocavuzojeq
Hastebin of Matrix's scores: https://hastebin.com/jozohiragu

Edit - New Spreadsheet showing some scores/players with pr59 (osu-performance) and pr2798 (osu) [high-AR and high-CS]: https://docs.google.com/spreadsheets/d/1fP8Fg2Rc8BELtoccAepu0TieJd26Tl1om0fRC_oFA-g/edit?usp=sharing

simo3003 commented:
Mind if you actually share the solution / source code for the calculator? Tried ILSpy and Telerik's JustDecompile to decompile it and attempt to fix the calculation of non-HD scores, but they don't seem to work.

peppy commented:
I've removed the links to binaries. Please link source code if necessary, not precompiled binaries.

Magnus-Cosmos commented:
Yeah, I I removed the entire text about the program, since it was quite broken in the first place. For the source code, 404 AimNotFound probably has it, I just decompiled the osu libraries that came with it and changed a few lines of code of osu-performance and osu-diffcalc, then recompiled and just bundled it with the program (prob very stupid way of going about that, but couldn't find an easy solution without requiring the source code).

Magnus-Cosmos commented:
Big thanks to Sakisan for making the hastebins:

Hastebin of filsdelama's top40 scores: https://hastebin.com/ocavuzojeq
Hastebin of Matrix's scores: https://hastebin.com/jozohiragu

Sakisan commented:
[Here is a page showing the new rankings](http://sakisan.be/osu/pr59.html)

Magnus-Cosmos commented:
Changed multiplier for aim to 1/3.4, as many people believed "nerfing" AR11 scores any more was unnecessary. Now, the rebalance mainly serves a buff to streaming with AR11 (which can be seen with maps like SoundTeMP - Dreamer's Dream).

Tom94 commented:
Can you please run the latest version of the code on the official DB dump (see the README file) and upload the pp table as a spreadsheet?
